[Doc] Pass on the contract chapter of the manual (#1999)
tweag/nickel: doc/manual/contracts.md

### Contracts parametrized by contracts

Contracts parametrized by other contracts are not really special amongst
parametrized contracts: it's just that the additional argument turns out to be a

> contract as well. Usually, a contract taking a contract parameter - say,
> `SubContract` - will most probably apply it eventually. You must use
> `std.contract.apply SubContract label value` or `std.contract.apply_as_custom
> SubContract label value` instead (the difference between both variants is
> explained in the next subsection).

Reviewer: ```suggestion
contract as well. Usually, a contract taking a contract parameter will eventually apply it, using `std.contract.apply` or `std.contract.apply_as_custom` (the difference between these is
explained in the next subsection).
```

> Parameters that are also contracts are generic. In particular, they are not
> necessarily predicates or validators but can contain delayed checks. In

Reviewer: ```suggestion
Parameters that are also contracts may contain delayed checks. In
```
